pkg/operator/staleconditions/remove_stale_conditions.go

```diff
@@ -63,55 +42,3 @@ func (c RemoveStaleConditions) sync() error {
 
 	return nil
 }
-
-// Run starts the kube-scheduler and blocks until stopCh is closed.
-func (c *RemoveStaleConditions) Run(ctx context.Context, workers int) {
-	defer utilruntime.HandleCrash()
-	defer c.queue.ShutDown()
-
-	klog.Infof("Starting RemoveStaleConditions")
-	defer klog.Infof("Shutting down RemoveStaleConditions")
-
-	if !cache.WaitForCacheSync(ctx.Done(), c.cachesToSync...) {
-		utilruntime.HandleError(fmt.Errorf("caches did not sync"))
-		return
-	}
-
-	// doesn't matter what workers say, only start one.
-	go wait.Until(c.runWorker, time.Second, ctx.Done())
```

Comment:
this is lost. The factory will run as many as the consumer asks for.

Reply:
`return factory.New().ResyncEvery(time.Second)` does this.

Reply:
we doing this for almost every controller....

Reply:
we have controllers which cannot run with multiple workers, yet the interfaces allow it, leading to races and worse things.